feat: add stop API to close active camera or media picker

NityaSantosh26 opened the pull request:
<!--
Please make sure the checklist boxes are all checked before submitting the PR. The checklist is intended as a quick reference, for complete details please see our Contributor Guidelines:

http://cordova.apache.org/contribute/contribute_guidelines.html

Thanks!
-->

### Platforms affected
Android and iOS


### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
I have an usecase where user opens camera and if there's an high priority push notification is received, when user clicks, it doesn't close the camera, User has to manually close the camera. Now using the navigator.camera.stop(). I can achieve this behavior.
And there are other instances which needs closing of camera or photo library from the code.


### Description
<!-- Describe your changes in detail -->
Added a new method stop() which handles the closing of CAMERA or PHOTOLIBRARY for both Android and iOS.


### Testing
<!-- Please describe in detail how you tested your changes. -->
I have verified the usecase I have mentioned above, when a push notification is clicked that closes the camera and takes to the destination page in the module by calling the stop() method.


### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

Commit:
Implement stop() to close CAMERA or PHOTOLIBRARY

NityaSantosh26 commented:
@breautek, My bad. I had an incomplete commit. Let me update my commit.

Commit:
updated stop() implementation

Commit:
Merge remote-tracking branch 'upstream/master'

breautek commented:
> Hey @breautek , Can I expect this PR also to be part of v8?

Sorry I was pre-occupied with more important changes that needed to make it's way through in v8 and I don't want to delay the release any further due to Android's October 31st deadline (the voting process takes minimum 48 hours so it is already behind) so I'll be beginning the vote process without this feature.

I bumped the milestone so that we can take a look for maybe an 8.1 release.

breautek reviewed (commented):
Our browser implementation for the camera I would say is barely usable but to make this feature complete, it would be nice to also have a browser implementation to hide the form input or whatever what the browser implementation does to show the camera/dialog.

GitToTheHub committed:
feat(camera): add stop API and complete native callbacks

- add navigator.camera.stop(success, error) to close active camera/photo picker
- handle stop action on Android and return success callback to JS
- avoid overwriting in-flight takePicture callback state when stop is called
- handle stop on iOS, dismiss picker on main thread, and always return success callback
- document stop API in README
- add stop to TypeScript definitions
- Generated-By: GPT-5.3-Codex

GitToTheHub committed:
Merge origin/master into PR-873 branch

GitToTheHub commented:
Hi @NityaSantosh26,

I mead some changes on you fork:
- Merged current code base
- handle stop action on Android and return success callback to JS
- handle stop on iOS, dismiss picker on main thread, and always return success callback
- document stop API in README
- add stop to TypeScript definitions

Generated-By: GPT-5.3-Codex

GitToTheHub commented:
@breautek is this good to go?

GitToTheHub committed:
fix(ios): make camera.stop robust during picker presentation

GitToTheHub committed:
Revert "fix(ios): make camera.stop robust during picker presentation"

This reverts commit a444000b5f1752c18c76a24a0179dc2ab8c39696.

GitToTheHub commented:
Tested it on iOS 26.5 on an iPhone 17 Pro simulator and Android 36.1 emulator (Pixel 10 Pro) by opening the camara to take a picture and closing it after 5 seconds, which worked.

Used JavaScript for test:

```js
window.testCameraStop = function () {
    navigator.camera.getPicture(
        function onSuccess(uri) { console.log("getPicture success:", uri); },
        function onError(err) { console.log("getPicture error:", err); },
        { sourceType: Camera.PictureSourceType.CAMERA, destinationType: Camera.DestinationType.FILE_URI }
    );

    setTimeout(function () {
        navigator.camera.stop(
            function () { console.log("stop success"); },
            function (err) { console.log("stop error:", err); }
        );
    }, 5000);
};

window.testCameraStop();
```
